docs/content/querying/virtual-columns.md

```diff
@@ -0,0 +1,59 @@
+---
+layout: doc_page
+---
+
+# Virtual Columns
+
+Virtual columns are queryable column "views" created from a set of columns during a query. 
+
+A virtual column can potentially draw from multiple underlying columns, although a virtual column always presents itself as a single column.
+
+Virtual columns can be used as dimensions or as inputs to aggregators.
+
+Each Druid query can accept a list of virtual columns as a parameter. The following scan query is provided as an example:
+
+```
+{
+ "queryType": "scan",
+ "dataSource": "page_data",
+ "columns":[],
+ "virtualColumns": [
+    {
+      "type": "expression",
+      "name": "fooPage",
+      "expression": "concat('foo' + page)",
+      "outputType": "STRING"
+    },
+    {
+      "type": "expression",
+      "name": "tripleWordCount",
+      "expression": "wordCount * 3",
+      "outputType": "LONG"
+    }
+  ],
+ "intervals": [
+   "2013-01-01/2019-01-02"
+ ] 
+}
+```
+
+## Expression virtual column
+
+Druid currently has one type of virtual column, the expression virtual column.
```

Comment:
We have one more virtual column type, `map` virtual column. This virtual column is implemented as an extension, so we need to add a separate page for `map` virtual column and add a link to http://druid.io/docs/latest/development/extensions.html.

Probably this doesn't have to be a part of this PR.

Reply:
The `map` one is from extensions-contrib, so I don't mention it here

Reply:
Ok, but `Druid currently has one type of virtual column` looks confused. How about `Druid basically supports the expression virtual column.`?

Reply:
I updated this part to have a "virtual column types" section header and put "expression" under there without additional comments